src/main/java/org/opensearch/security/filter/SecurityRestFilter.java

```diff
@@ -138,6 +145,42 @@ private boolean userIsSuperAdmin(User user, AdminDNs adminDNs) {
         return user != null && adminDNs.isAdmin(user);
     }
 
+    private boolean authorizeRequest(RestHandler original, RestRequest request, RestChannel channel, User user) throws Exception {
+        if (original instanceof RestSendToExtensionAction) {
+            List<RestHandler.Route> extensionRoutes = original.routes();
+            Optional<RestHandler.Route> handler = extensionRoutes.stream()
+                    .filter(rh -> rh.getMethod().equals(request.method()))
+                    .filter(rh -> restPathMatches(request.path(), rh.getPath()))
+                    .findFirst();
+            if (handler.isPresent() && handler.get() instanceof NamedRoute) {
+                String action = ((NamedRoute)handler.get()).name();
+                PrivilegesEvaluatorResponse pres = evaluator.evaluate(user, action);
+                if (log.isDebugEnabled()) {
+                    log.debug(pres.toString());
+                }
+
+                if (pres.isAllowed()) {
+                    // TODO make sure this is audit logged
+                    log.debug("Request has been granted");
```

Comment:
Is there a performance impact of running these checks against rest + transport actions?  If unclear, how about creating a new issue on the meta.

Reply:
done